src/core/lib/slice/slice_buffer.c

```diff
@@ -273,7 +306,8 @@ void grpc_slice_buffer_move_first(grpc_slice_buffer *src, size_t n,
       grpc_slice_buffer_add(dst, slice);
       break;
     } else { /* n < slice_len */
-      grpc_slice_buffer_undo_take_first(src, grpc_slice_split_tail(&slice, n));
+      grpc_slice_buffer_undo_take_first(
+          src, grpc_slice_split_tail_no_ref(&slice, n));
```

Comment:
Same comment as above that this duplicates too much code. given this use, I think adding a bool parameter that determines if refcounts will be altered is the cleanest fix, then it can simply be passed to grpc_slice_split_tail with not more changes needed.